webapp/src/js/controllers/analytics-targets.js

```diff
@@ -12,19 +9,23 @@ angular.module('inboxControllers').controller('AnalyticsTargetsCtrl',
     const ctrl = this;
 
     ctrl.targets = [];
-    ctrl.targetsDisabled = !RulesEngine.enabled;
     ctrl.loading = true;
+    ctrl.targetsDisabled = false;
 
-    TargetGenerator(function(err, targets) {
-      if (err) {
-        return $log.error('Error fetching targets', err);
-      }
-      // timeout to force digest
-      $timeout(function() {
+    RulesEngine.isEnabled()
+      .then(isEnabled => {
+        ctrl.targetsDisabled = !isEnabled;
+        ctrl.loading = isEnabled;
+        return isEnabled ? RulesEngine.fetchTargets() : [];
+      })
+      .then(targets => {
+        ctrl.loading = false;
         ctrl.targets = targets;
+      })
+      .catch(err => {
+        $log.error('Error getting targets', err);
         ctrl.loading = false;
+        ctrl.targets = [];
```

Comment:
I'd move the `catch` up to line 21 and log the error and return `[]`. Then the `then` works like a `finally`. eg:

```js
  ...
  return isEnabled ? RulesEngine.fetchTargets() : [];
})
.catch(err => {
  $log.error('Error getting targets', err);
  return [];
})
.then(targets => {
  ctrl.targets = targets;
  ctrl.loading = false;
});
```

Reply:
:+1: